### PR TITLE
[Bug] Plugin content locking (referrer POST) causes 500.

### LINKED ISSUE #5787: [Bug] Plugin content locking (referrer POST) causes 500.
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Yes
| Feature request? | No
| Enhancement? | No

## Description:
When two users attempt to edit a Focus at the same time, one will receive a 500 error due to content locking. This is a core bug within `mautic/app/bundles/CoreBundle/Controller/AbstractFormController.php` which is not currently cognizant of plugins.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.12.2
| PHP version | all

### Steps to reproduce:
1. Edit a Focus
2. Ask your friend to try to edit the same Focus while you have it checked out.
 
### Log errors: 
```
 mautic.CRITICAL: Uncaught PHP Exception InvalidArgumentException: "The "MauticPluginMauticFocusBundle" (from the _controller value "MauticPluginMauticFocusBundle:MauticFocus:view") does not exist or is not enabled in your kernel!" at /var/app/current/mautic/app/cache/prod/classes.php line 2888 {"exception":"[object] (InvalidArgumentException(code: 0): The \"MauticPluginMauticFocusBundle\" (from the _controller value \"MauticPluginMauticFocusBundle:MauticFocus:view\") does not exist or is not enabled in your kernel! at /var/app/current/mautic/app/cache/prod/classes.php:2888, InvalidArgumentException(code: 0): Bundle \"MauticPluginMauticFocusBundle\" does not exist or it is not enabled. Maybe you forgot to add it in the registerBundles() method of your AppKernel.php file? at /var/app/current/vendor/symfony/http-kernel/Kernel.php:228)"} []
```

### PR DESCRIPTION
**Pull request by heathdutton**
| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #5789
| BC breaks? | No
| Deprecations? | 

[//]: # ( Required: )
#### Description:
When a redirection should occur by ajax (such as "permission denied" or "content locked" situations) and the user is using a plugin (Focus plugin for example), the user may encounter a 500 error. The `referrrerPostAction` cannot find the class intended to handle the action. A minor change should fix this.

#### Steps to test this PR:
1. Attempt to edit the same Campaign (core) as two different users, no 500 error should occur.
2. Attempt to edit the same Focus (plugin) as two different users, no 500 error should occur.

### CODE REVIEW

File: app/bundles/CoreBundle/Controller/AbstractFormController.php [hunk 1]
```diff
@@ -237,19 +237,17 @@ protected function copyErrorsRecursively(Form $copyFrom, Form $copyTo)
...
     protected function refererPostActionVars($vars)
     {
         if (empty($this->request->server->get('HTTP_REFERER'))) {
             return $vars;
         }
 
-        $returnUrl                              = !empty($this->request->server->get('HTTP_REFERER'))
-                                                ? $this->request->server->get('HTTP_REFERER')
-                                                : $returnUrl;
+        $returnUrl                              = $this->request->server->get('HTTP_REFERER');
```
Comment (dbhurley) [inline app/bundles/CoreBundle/Controller/AbstractFormController.php]: This change does not test for the value of the request existing before setting the value.

Comment (heathdutton) [reply app/bundles/CoreBundle/Controller/AbstractFormController.php]: Also, $returnUrl isn't defined.

Comment (heathdutton) [reply app/bundles/CoreBundle/Controller/AbstractFormController.php]: Please take another look. This line was faulty to begin with as $returnUrl is never defined, and line 246 already checks for an empty referrer.

**Comment by heathdutton** [comment]
We've been running this in production as a patch for all plugins for some time, as we have a few concurrent users and they were hitting the 500 error frequently for Focus and other third-party plugins whenever trying to edit the same entity. Just need to have this confirmed by another party :)

**Comment by escopecz** [comment]
@heathdutton can you fix the conflicts please?

**Comment by escopecz** [comment]
@heathdutton I cannot reproduce by following the steps.

Can you guide me how to reproduce it? I did this:
1. Create another admin user
2. Edit existing focus item
3. Open incognito browser
4. Log in to Mautic as the second admin user
5. Edit the same focus item.

This is the result:
screen shot 2018-10-08 at 20 16 15

**Comment by escopecz** [comment]
@heathdutton are my steps correct? How to replicate?

**Pull request closed without merging**